pkg/daemon/cleanup/cleanup.go

```diff
@@ -102,6 +72,77 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 	return nil
 }
 
+// getServicesToRemove returns a list of services that should be removed as part of the cleanup process
+// returns the merged Data of the latest ConfigMap, and the ConfigMap specified by the node's version annotation
+func getServicesToRemove(ctx context.Context, cli client.Client, node *core.Node, configMapNamespace string) ([]servicescm.Service, error) {
+	// get data from the latest services ConfigMap
+	cm, err := servicescm.GetLatest(cli, ctx, configMapNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get latest services ConfigMap from namespace %s: %w", configMapNamespace, err)
+	}
+	cmData, err := servicescm.Parse(cm.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	// attempt to get the ConfigMap specified by the version annotation
+	var versionCM core.ConfigMap
+	var versionCMData *servicescm.Data
+	err = func() error {
+		if node == nil {
+			return fmt.Errorf("no node object")
+		}
+		version, present := node.Annotations[metadata.VersionAnnotation]
+		if !present {
+			return fmt.Errorf("node is missing version annotation")
+		}
+		err = cli.Get(ctx, client.ObjectKey{Namespace: configMapNamespace, Name: servicescm.NamePrefix + version},
+			&versionCM)
+		if err != nil {
+			return err
+		}
+		versionCMData, err = servicescm.Parse(versionCM.Data)
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+	if err != nil {
+		klog.Infof("error getting services ConfigMap associated with version annotation, "+
```

Comment:
do you think it's worth trying to get the CM associated with the desiredVersion annotation in this case? or is that most likely the most recently created services CM anyway

Reply:
Can you refresh in what case are the VersionAnnotation and DesiredVersionAnnotation different?